providers/implementations/signature/rsa.c

```diff
@@ -728,7 +812,7 @@ static int rsa_digest_signverify_init(void *vprsactx, const char *mdname,
     if (prsactx != NULL)
         prsactx->flag_allow_md = 0;
     if (!rsa_signverify_init(vprsactx, vrsa, operation)
```

Comment:
The flow here is a little haphazard currently...
rsa_signverify_init() internally calls rsa_setup_md() and then it also
calls it here after the init and redoes it even when the name is the same.. Maybe line 276 should do a compare?

Reply:
Added check directly here